src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -316,6 +338,26 @@ public void transferItemHold(UUID transactionId, String itemBarcode) {
     eventPublisher.publishEvent(MoveRequestEvent.of(transaction, item));
   }
 
+  private void recallItem(InnReachTransaction transaction, UUID itemId) {
+    List<RequestDTO> requestList = requestService.getRequestsByItemId(itemId);
+    for (RequestDTO request : requestList) {
+      if (request.getStatus() == OPEN_NOT_YET_FILLED) {
+        transaction.setState(RECALL);
+        var loan = loanService.getById(transaction.getHold().getFolioLoanId());
+        var loanDueDate = loan.getDueDate().toInstant().truncatedTo(ChronoUnit.SECONDS);
+        var loanIntegerDueDate = (int) (loanDueDate.getEpochSecond());
+        transaction.getHold().setDueDateTime(loanIntegerDueDate);
+        notifier.reportRecallRequested(transaction, loanDueDate);
```

Comment:
adjust and re-use existing method `updateTransactionOnLoanRecallRequested`

Reply:
Fixed

Reply:
Fixed